backend/api/views.py

```diff
@@ -0,0 +1,364 @@
+    list_of_intents = intents_json['intents']       
+
+    for i in list_of_intents:                       
+        if i['tag'] == tag:                         
+            result = random.choice(i['responses'])  
+            break                                   
+    
+    return result                                   
+
+
+def capture_and_recognize(request):
+    if request.method == 'GET':
+        r = sr.Recognizer()
+        with sr.Microphone() as source:
+            audio_text = r.listen(source)
+            try:
+                # Using Google speech recognition
+                text_en = r.recognize_google(audio_text)
+                # Adding French language option
+                text_fr = r.recognize_google(audio_text, language="fr-FR")
+                # Adding Spanish language option
+                text_es = r.recognize_google(audio_text, language="es-AR")
+
+                return (text_en, text_fr, text_es)
+
+            except sr.UnknownValueError:
+                return HttpResponse("Could not understand audio. Please press the button again to try again!")
+            
+            except sr.RequestError as e:
+                return HttpResponse(f"Could not request results from Google Speech Recognition service; {e}")
+            
+    return HttpResponse("Method Not Allowed")
+
+
+def lang_detect(response1, response2, response3):
+    language_prob = []
+
+    # Make into string elements for data cleaning
+    detect_en = [str(i) for i in detect_langs(response1)]
+    language_prob.extend(detect_en)
+
+    detect_fr = [str(i) for i in detect_langs(response2)]
+    language_prob.extend(detect_fr)
+
+    detect_es = [str(i) for i in detect_langs(response3)]
+    language_prob.extend(detect_es)
+
+    return list(language_prob)
+
+
+def lang_prob(lang_probability):
+    language_code = []
+
+    for i in range(0, len(lang_probability)):         
+        code = lang_probability[i][0:2]
+        language_code.append(code)
+
+        lang_probability[i] = lang_probability[i][3:]
+    
+    probs = [float(i) for i in lang_probability]
+
+    return language_code, probs
+
+
+def ISO_639(langauge_code, probability):
+    lang_counter = Counter(langauge_code)
+
+    test_val = list(lang_counter.values())[0]
+    res = True
+    
+    for i in lang_counter:
+        if lang_counter[i] != test_val:
+            res = False
+            break
+
+    if res == True:
+        d = defaultdict(list)
+        for key, value in zip(langauge_code, probability):
+            d[key].append(value)
+
+        mean_probs = {key: np.mean(val) for key, val in dict(d).items()}
+
+        ISO_639_2 = str(max(mean_probs, key = mean_probs.get))
+
+    else:
+        ISO_639_2 = max(lang_counter, key = lang_counter.get)
+    
+    return ISO_639_2
+
+
+@api_view(["GET"])
+# When the button is pressed
+def main(request):
+
+    intents = load_intents()
+
+    English, French, Spanish = capture_and_recognize(request)
```

Comment:
Sometimes if the audio wasn't captured properly an exception gets thrown here, can we please look at handling that more gracefully? Maybe use a try except block to catch that exception

![image](https://github.com/Project-BabelBot/BabelBot-Code/assets/70828477/3eaceabf-f1d3-4256-93eb-ac524771d504)

Reply:
Updated in the push in my recent push.